### ISSUE TITLE
ConvNoBias does not support shape inference for unknown dimension

### ISSUE THREAD

**Issue by tungld** [clang-ykt/ONNF#107]
The following excerpt is from mnist model:
```mlir
```

**Comment by doru1004** [comment]
I'm not sure why you say it's wrong, there's no way to infer the other dimensions. Can you please explain what you mean?

**Comment by tungld** [comment]
I see. Thanks for the explanation!

When Conv's shape inference produces an unknown dimension type, I think the lowering code for Conv should dynamically compute values for unknown dimensions in order to allocate memory. It seems, the current lowering code for Conv hasn't handle this situation, has it?

**Comment by doru1004** [comment]
You're right. Conv doesn't handle that dynamically yet. To handle it, you would have to emit all the computations that are currently being done inside the shape inference function.
